### ISSUE TITLE
history.NextAsync() throws 40000 exception

### ISSUE THREAD

**Issue by ndastur** [ably/ably-dotnet#352]
Using ably 1.1.8 SDK

Steps to reproduce

1) Publish 100+ messages on a channel
2) Use this code to get a history
```
var historyParams = new PaginatedRequestParams()
{
   Direction = QueryDirection.Forwards,
   Start = Settings.MessageGatewayLastMessageInTimestamp
                                        .FromUnixTimeInMilliseconds()           // get the DateTimeOffset
                                        .AddMinutes(-2)                        // go back sometime to capture all
};

   PaginatedResult<Message> history;
   var specific_incoming_chat_channel = RealtimeObj.Channels.Get(specificChannelName);
   history = await specific_incoming_chat_channel.HistoryAsync(historyParams);

   // Loop over the history
   if (history == null)
       return null;

   do
                    {
                        Debug.WriteLine($"{Tag} got history object with {history.Items.Count} items");

                        if (cancellationToken.IsCancellationRequested)
                            cancellationToken.ThrowIfCancellationRequested();

                        if (!history.HasNext)
                            break;

                        history = await history.NextAsync();
                    }
   while (true);
```

This will throw an exception at ```history = await history.NextAsync();```
Exception: [ErrorInfo Reason: Invalid mode specification. (See https://help.ably.io/error/40000 for help.); Code: 40000; StatusCode: 400 (BadRequest); Href: https://help.ably.io/error/40000]

**Comment by paddybyers** [comment]
This is the request we're receiving:

```
"GET /channels/chat%3AR%3Ahistory_test_1ac4e3722459c/messages?start=946684680000&end=1561853924942&direction=forwards&limit=100&cont=true&format=json&firstStart=946684680000&fromDate=946684680000&mode=all%3E%3B+rel%3D%22first%22%2C+%3C.%2Fmessages"
```

**Comment by ndastur** [comment]
Glad the problem is identified. At the moment this blocks our production app getting me than 100 messages from the history.

**Comment by paddybyers** [comment]
> At the moment this blocks our production app getting me than 100 messages from the history.

If it helps, you can specify a `limit` param to be any value up to 1000 and get that number of messages in the first page.

**Comment by ndastur** [comment]
Okay. Thanks

**Comment by withakay** [comment]
This will be fixed in the 1.1.20 release due today

**Comment by ndastur** [comment]
Thanks @withakay . In our test app the fix works for Android and iOS
